_test/test_mpi/test_CPP_MPI_exchange.m

```diff
@@ -13,24 +13,226 @@
             end
             obj = obj@TestCase(name);
         end
-        function xest_MessagesCppMPI_constructor(this)
-            % crashes Matlab if applied for second time
-            % needs some thinking on how to avoid this. 
+        function test_JobExecutor(obj)
             if isempty(which('cpp_communicator'))
                 return
             end
-            mf = MessagesCppMPI_tester('test_comm');
+            serverfbMPI  = MessagesFilebased('test_JE_CppMPI');
+            serverfbMPI.mess_exchange_folder = tmp_dir;
+            
+            [data_exchange_folder,JOB_id] = fileparts(serverfbMPI.mess_exchange_folder);
+            cs = iMessagesFramework.build_worker_init(fileparts(data_exchange_folder),...
+                JOB_id,'MessagesCppMPI_test3',1,3,true);
+            
+            % intercomm constructor invoked here.
+            [fbMPI,intercomm] = JobExecutor.init_frameworks(cs);
+            clob1 = onCleanup(@()(finalize_all(intercomm)));
+            clob2 = onCleanup(@()(finalize_all(fbMPI)));
+            assertTrue(isa(intercomm,'MessagesCppMPI'));
+            assertEqual(intercomm.labIndex,uint64(1));
+            assertEqual(intercomm.numLabs,uint64(3));
+            
+            
+            % send fake messages, presumably generated by workers 2 and 3
+            m1 = aMessage('started');
+            m1.payload = 'a';
+            [ok,err_mess] = intercomm.send_message(2,m1);
+            assertEqual(ok,MESS_CODES.ok,['Error: ',err_mess]);
+            m1.payload = 'b';
+            [ok,err_mess] = intercomm.send_message(3,m1);
+            assertEqual(ok,MESS_CODES.ok,['Error: ',err_mess]);
+            
+            
+            je = JETester();
+            common_job_param = struct('filepath',data_exchange_folder,...
+                'filename_template','test_jobDispatcherL%d_nf%d.txt',...
+                'fail_for_labsN',2:3);
+            im = InitMessage(common_job_param,1,1,1);
+            
+            je = je.init(fbMPI,intercomm,im,true);
+            
+            
+            [ok,err_mess,message] = serverfbMPI.receive_message(1,'started');
+            assertEqual(ok,MESS_CODES.ok,['Error: ',err_mess]);
+            assertEqual(message.mess_name,'started')
+            assertEqual(numel(message.payload),3);
+            assertTrue(isempty(message.payload{1}));
+            assertEqual(message.payload{2},'a');
+            assertEqual(message.payload{3},'b');
+            
+        end
+        function test_SendReceive(obj)
```

Comment:
is this one test or lots of tests?

Reply:
the keyword is "test_" this is how the framework identify its cases. At least that what I beliveve in

Reply:
I mean is this one test function testing one thing or testing many things?

Reply:
it is testing send-receive operation or rather correct matlab send-receive interface as nothing is actually send/received here.